src/Advection/weno_fifth_order.jl

```diff
@@ -177,17 +177,27 @@ Adapt.adapt_structure(to, scheme::WENO5{FT, XT, YT, ZT, XS, YS, ZS, W}) where {F
         stencil = retrieve_left_smooth(scheme, r, dir, i, location)
         wᵢᵢ = stencil[1]   
         wᵢⱼ = stencil[2]
+        # horrible but have to do this for GPU execution (broadcast doesn't work apparently)
```

Comment:
You can't broadcast _inside_ a kernel. Broadcasting can _launch_ a kernel. Everything inside a kernel has to compile to CUDA C. I didn't know that the compile could unroll loops like what you've written though, that's awesome!

Reply:
Ah cool! Launching a kernel with a broadcast seems like a nice feature! (Tool me a while to figure it out though:) ) Then you mean that while running on the GPU, threads at lines 152 -154 are launching their own child dynamic kernel with size 3? Or is the compiler handling it in some weird way?

Regarding the for loops from what I know there is nothing preventing them in CUDA C